src/views/v3/Bridge/AssetPicker/ChainList.tsx

```diff
@@ -0,0 +1,231 @@
+import React, { useMemo, useState } from 'react';
+import { useTheme } from '@mui/material/styles';
+import Card from '@mui/material/Card';
+import CardContent from '@mui/material/CardContent';
+import List from '@mui/material/List';
+import ListItemButton from '@mui/material/ListItemButton';
+import ListItemIcon from '@mui/material/ListItemIcon';
+import Stack from '@mui/material/Stack';
+import Tooltip from '@mui/material/Tooltip';
+import Typography from '@mui/material/Typography';
+
+import ChainIcon from 'icons/ChainIcons';
+import PlusIcon from 'icons/Plus';
+
+import type { ChainConfig } from 'config/types';
+import type { WalletData } from 'store/wallet';
+import SearchableList from 'views/v3/Bridge/AssetPicker/SearchableList';
+
+import { Chain } from '@wormhole-foundation/sdk';
+
+type Props = {
+  chainList?: ChainConfig[];
+  selectedChainConfig?: ChainConfig;
+  showSearch: boolean;
+  setShowSearch: (value: boolean) => void;
+  wallet: WalletData;
+  onChainSelect: (chain: Chain) => void;
+};
+
+const SHORT_LIST_SIZE = 5;
+
+const ChainList = (props: Props) => {
+  const theme = useTheme();
+  const [chainSearchQuery, setChainSearchQuery] = useState('');
+
+  const styles = useMemo(
+    () => ({
+      card: {
+        background: theme.palette.input.background,
+        maxWidth: '420px',
+        [theme.breakpoints.down('sm')]: {
+          width: '100vw',
+        },
+      },
+      cardContent: {
+        paddingBottom: '0!important',
+        [theme.breakpoints.down('sm')]: {
+          padding: '16px 10px',
+          ':last-child': {
+            padding: '16px 10px',
+          },
+        },
+      },
+      title: {
+        fontSize: '14px',
+        marginBottom: '12px',
+      },
+      chainSearch: {
+        maxHeight: '400px',
+        [theme.breakpoints.down('sm')]: {
+          maxHeight: '600px',
+        },
+      },
+      chainButton: {
+        display: 'flex',
+        flexDirection: 'column' as const,
+        padding: '8px',
+        border: '1px solid transparent',
+        borderRadius: '8px',
+        '&.Mui-selected': {
+          border: '1px solid',
+          borderColor: theme.palette.primary.main,
+        },
+      },
+      chainItem: {
+        display: 'flex',
+        flexDirection: 'row' as const,
+        padding: '8px',
+        borderRadius: '8px',
+      },
+    }),
+    [theme],
+  );
+
+  const {
+    chainList,
+    selectedChainConfig,
+    showSearch,
+    setShowSearch,
+    onChainSelect,
+  } = props;
+
+  const [topChains, showMoreButton] = useMemo(() => {
```

Comment:
showMoreButton seems = allChains.length > SHORT_LIST_SIZE, doesn't need to be memoized